tests/fixtures/extend-expect-circuit-snapshot.ts

```diff
@@ -29,20 +30,187 @@ async function saveSvgSnapshotOfCircuitJson({
   const filePath = path.join(snapshotDir, snapshotName)
 
-  let svg: string
+  let content: Buffer | string
   switch (mode) {
     case "pcb":
-      svg = convertCircuitJsonToPcbSvg(soup)
+      content = convertCircuitJsonToPcbSvg(soup)
       break
     case "schematic":
-      svg = convertCircuitJsonToSchematicSvg(soup, options)
+      content = convertCircuitJsonToSchematicSvg(soup, options)
       break
-    case "simple-3d":
-      svg = await convertCircuitJsonToSimple3dSvg(soup, options)
+    case "simple-3d": {
+      // Convert circuit-json to glTF/GLB, then render to PNG with poppygl
+      const gltfModule: any = await import("circuit-json-to-gltf")
+      const toGltf =
+        gltfModule.convertCircuitJsonToGltf ?? gltfModule.circuitJsonToGltf
+      if (!toGltf) {
+        throw new Error(
+          "circuit-json-to-gltf does not export convertCircuitJsonToGltf or circuitJsonToGltf",
+        )
+      }
+      const gltfOrGlb = await toGltf(soup, options?.gltf)
+
+      const poppy: any = await import("poppygl")
+      const {
+        renderGLTFToPNGBufferFromGLBBuffer,
+        bufferFromDataURI,
+        createSceneFromGLTF,
+        decodeImageFromBuffer,
+        computeWorldAABB,
+        pureImageFactory,
+        renderSceneFromGLTF,
+        encodePNGToBuffer,
+      } = poppy
+
+      // If we got a GLB buffer, render directly
+      if (gltfOrGlb instanceof Uint8Array || Buffer.isBuffer(gltfOrGlb)) {
+        const glbBuffer = Buffer.isBuffer(gltfOrGlb)
+          ? gltfOrGlb
+          : Buffer.from(gltfOrGlb)
+        const resolvedRenderOpts = {
+          width: 1024,
+          height: 1024,
+          ambient: 0.2,
+          gamma: 2.2,
+          ...(options?.poppygl ?? {}),
+        }
+        const png = await renderGLTFToPNGBufferFromGLBBuffer(
+          glbBuffer,
+          resolvedRenderOpts,
+        )
+        if (process.env.SAVE_3D_DEBUG_SNAPSHOT === "1") {
+          const debugPath = filePath.replace(/\.png$/, ".glb")
+          fs.writeFileSync(debugPath, glbBuffer)
+        }
+        content = Buffer.isBuffer(png) ? png : Buffer.from(png)
+        break
+      }
+
+      // Otherwise assume GLTF JSON (object or JSON string)
+      const gltfJson =
+        typeof gltfOrGlb === "string" ? JSON.parse(gltfOrGlb) : gltfOrGlb
+
+      // Prepare resources; require data: URIs to avoid filesystem/network in tests
+      const buffers = await Promise.all(
+        (gltfJson.buffers ?? []).map(async (entry: any) => {
+          if (!entry?.uri) {
+            throw new Error(
+              "GLTF buffer without a URI; please configure circuit-json-to-gltf to emit GLB, or ensure buffers use data: URIs.",
+            )
+          }
+          if (!entry.uri.startsWith("data:")) {
+            throw new Error(
+              `Non-data URI buffer (${entry.uri}) not supported in tests; use GLB or embed resources.`,
+            )
+          }
+          return bufferFromDataURI(entry.uri)
+        }),
```

Comment:
What is happening here, this doesnt belong in core.circuit json to gltf should produce a glb file which wouldnt require resource embedding